### PR TITLE
Elasticsearch: Disable Alias field for non-timeseries queries

### LINKED ISSUE #27521: Elasticsearch: Disable alias field for queries that do not have date histogram
**What happened**: 
Alias is not working without Date Histogram in query

**What you expected to happen**: 
Alias work without time series is set or not

**How to reproduce it (as minimally and precisely as possible)**:
Alias is not working without Date Histogram
vgca-01

But if i set Date Histogram, alias worked
vgca-02

**Anything else we need to know?**:
I try to join/merge beetween 2 query in to single table with field "org.L", but two query have same field "Count" (alias is not worling) so join/merge transformations is wrong. It show two column (org.L, count) instead of (org.L, Total A, Total B)

**Environment**:
- Grafana version: 7.1.5
- Data source type & version: ES 7.9.0
- OS Grafana is installed on: Centos 7
- User OS & Browser: Chrome latest
- Grafana plugins: Table Panel
- Others:

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:
The Alias field only makes sense when processing timeseries queries. This PR adds a tooltip to the field explaining this and disables the field when the last bucket aggregation in the query is not a Date Histogram.

**Which issue(s) this PR fixes**:

Fixes #27521

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx [hunk 1]
```diff
@@ -45,5 +45,41 @@ describe('QueryEditor', () => {
...
+            id: '1',
+            type: 'avg',
+          },
+        ],
+        bucketAggs: [{ id: '2', type: 'date_histogram' }],
+      };
+
+      render(<QueryEditor query={query} datasource={{} as ElasticDatasource} onChange={noop} onRunQuery={noop} />);
+
+      expect(screen.getByLabelText('Alias')).not.toBeDisabled();
```
Comment (Elfo404) [inline public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx:82]: just a  (personal) nit:

```
      expect(screen.getByLabelText('Alias')).not.toBeDisabled();
```

or

```
      expect(screen.getByLabelText('Alias')).toBeEnabled();
```
?

Comment (zoltanbedi) [reply public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx:82]: I would probably go with toBeEnabled but this is just personal preference.

**Review by zoltanbedi — APPROVED** [review]
This looks good to me. Just one question, if the user changes the alias field and then changes other fields so it gets disabled is the changes that are still there going to matter?

**Comment by Elfo404** [comment]
@zoltanbedi no, for non-timeseries queries that fields get completely ignored, that's why I specified it in the tooltip.
>For all other query types this field is ignored.

I actually wanted to remove the field completely, but as @gabor pointed out it might be disturbing to see it appearing and disappearing.

do you think it's clear enough or would you change something?

**Comment by zoltanbedi** [comment]
> do you think it's clear enough or would you change something?

I think it is clear.

File: public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx (hunk 1 above)
Comment (Elfo404) [reply public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx:82]: will update! thanks! ❤️

Comment (Elfo404) [inline public/app/plugins/datasource/elasticsearch/components/QueryEditor/index.test.tsx:82]:
```suggestion
      expect(screen.getByLabelText('Alias')).toBeEnabled();
```

**Pull request merged**